File: Semester 1/ALGOLAB1/taskG.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
  int t, n, s;
  string q;
  cin >> t;
  for (int i = 0; i < t; i++) {
    cin >> n;
    int votes[n];
    int minimum = 15;
    int maximum = 30;
    for (int j = 0; j < n; j++) {
      cin >> q >> s;
      votes[j] = s;
      if (q == ">=" and minimum < s) {
        minimum = s;
      }
      if (q == "<=" and maximum > s) {
        maximum = s;
      }
      if (minimum <= s <= maximum and maximum >= minimum) {
        cout << minimum << "\n";
      } else {
        cout << -1 << "\n";
      }
    }
  }
  return 0;
}

File: Semester_1/ALGOLAB1_Python/taskG.py
def main():
    t = int(input())
    for _ in range(t):
        n = int(input())
        minimum = 15
        maximum = 30
        for _ in range(n):
            q, s = input().split()
            s = int(s)
            if q == ">=" and minimum < s:
                minimum = s
            if q == "<=" and maximum > s:
                maximum = s
            if minimum <= maximum:
                print(minimum)
            else:
                print(-1)

if __name__ == "__main__":
    main()
